app/models/token_authentication.rb

```diff
@@ -0,0 +1,13 @@
+require 'jwt'
+
+# Not an ActiveRecord; not instantiated, actually
```

Comment:
The standard practice, if you will never instantiate a class, is to make it a `module` and then `extend` (for class methods) or `include` (for instance methods) your module as needed. So in this case, you could call `TokenAuthentication` a `module` and, above in `ServicesController`, you could `extend TokenAuthentication`, and you should have access to these two class methods "natively" (no need to call TokenAuthentication specifically).

Also, when you have more than one class method being declared, it is stylistically preferable to use this syntax:
```ruby
class Foo
  class << self
    def first_class_method
      # whatever
    end
    def second_class_method
      # more stuff
    end
  end
end
```

...instead of the `def self.method` syntax.

Reply:
So what directory is this utility supposed to live in? One source suggested creating a `lib/` directory but I didn't want to disrupt the current structure.

And `ServicesController` is already a subclass of `ApplicationController` - how can it extend two things?...  I guess I should read up on ruby classes and modules although so far I've only seen tutorial like presentations, nothing that actually explains how they work.

I've seen that class-in-class (singleton?) business before but don't want to use it until I understand it.

Reply:
There are two schools of thought for where to put files. The older (and arguably more popular one) is that `/app/models` is for relational models stored in the database, and `/lib` is for everything else. The other, which I prefer and am using here, is that `/app/models` are for classes specific to your Rails project, and `/lib` is for more general classes that could (really "should") be extracted into their own projects. [shrug] It's not a huge problem to use one or the other.

Right, `ServicesController` is already a subclass of `ApplicationController`. It will contnue to be. the use of `extend` does NOT imply inheritance (e.g.: subclassing) of any kind. `extend` means, roughly, "steal the class methods (and variables) from this other object". If you're a Java sort, you can think of a module as a kind of abstract class and `extend` as some version of `implements`. ...Kind of. :) Ruby is a little more loosey-goosey in this domain. :D

There are no singletons involved, here, it's just a way of referencing methods from another class. ...A separation of concerns.

TL;DR: Calling it a "class" instead of a "module" isn't strictly *wrong*, but stylistically it's preferable to make it a `module` and to use `extend`. Don't sweat it if the change is intimidating.